torch/distributed/tensor/_ops/_tensor_ops.py

```diff
@@ -286,6 +287,81 @@ def gen_slice_strategy(op_schema: OpSchema) -> StrategyType:
     return slice_strategy
 
 
+@register_op_strategy(
+    torch.ops.aten.select.int,
+    schema_info=RuntimeSchemaInfo(1),
+)
+def select_int_strategy(mesh: DeviceMesh, op_schema: OpSchema) -> OpStrategy:
+    # func: select.int(Tensor(a) self, int dim, SymInt index) -> Tensor(a)
+    args_schema = op_schema.args_schema
+    input_strategy, dim = args_schema[0], args_schema[1]
+    assert isinstance(input_strategy, OpStrategy), f"{input_strategy=}"
+    assert isinstance(dim, int), f"{dim=}"
+    output_strategies: list[PlacementStrategy] = []
+    for placement_strategy in input_strategy.strategies:
+        output_spec = placement_strategy.output_spec
+        new_placements = []
+        for placement in output_spec.placements:
+            if isinstance(placement, Shard) and placement.dim == dim:
+                # Indexing into the sharded dim requires replication
+                new_placements.append(Replicate())
+            elif isinstance(placement, Shard):
+                shard_dim = placement.dim
+                if dim < shard_dim:
+                    # If the dim being selected is less than the shard dim,
+                    # then we must decrement the shard dim since the selected
+                    # dim is now removed
+                    new_placements.append(Shard(shard_dim - 1))
+                else:
+                    new_placements.append(Shard(shard_dim))
+            else:
+                new_placements.append(placement)
+        assert isinstance(output_spec.tensor_meta, TensorMeta)
+        new_size = list(output_spec.tensor_meta.shape)
+        new_size[dim] = 1
+        tensor_meta = TensorMeta(torch.Size(new_size), (0,), dtype=output_spec.tensor_meta.dtype)
```

Comment:
usually you don't need to manually compute the output shape/stride here because fake tensor propagation would try to do shape propagation after the sharding propagation. Is this the issue you mentioned about the cache miss?